server/data/data-ldap/src/test/java/org/apache/james/user/ldap/ReadOnlyUsersLDAPRepositoryTest.java

```diff
@@ -69,188 +73,620 @@ static void afterAll() {
+        @Test
+        public void addUserShouldDisableCaseVariationWhenOriginalValueLowerCased(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void testShouldReturnTrueWhenAUserHasAnIncorrectCaseName(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void addUserShouldAddAUserWhenEmptyRepository(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void testShouldBeCaseInsentive(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void containsShouldBeCaseInsentive(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void containsShouldPreserveCaseVariation(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void updateUserShouldNotAllowToAuthenticateWithOldPassword(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void getUserByNameShouldReturnAUserWhenContainedInRepository(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void removeUserShouldRemoveAUserWhenPresentInRepository(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void testShouldReturnFalseWhenAUserHasAnIncorrectPassword(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void testShouldReturnFalseWhenAUserIsRemovedFromRepository(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void removeUserShouldBeCaseInsentiveOnCaseVariationUser(TestSystem testSystem) {
+        }
+
+        @Disabled("JAMES-3088 This user-repository is read-only. Modifications are not permitted.")
+        @Override
+        @Test
+        public void listShouldReturnLowerCaseUser(TestSystem testSystem) {
```

Comment:
it is not write operation. No ?

Reply:
to test list(), the test case added some user to repository. That's why

Reply:
BTW I had rewritten the change set, this change is not available now